packages/auth-core/src/core.ts

```diff
@@ -145,6 +145,13 @@ export class Auth {
     });
   }
 
+  async resendVerificationEmailForEmail(email: string) {
```

Comment:
I thought a bit about how we might be able to support this format directly on `resendVerificationEmail`, but I think I prefer the direction you've taken this here by making it just a separate function. Moves the complexity into the consumer.